src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -126,7 +132,7 @@ public ISymbolSlot AddVariable(string name, FormulaType type, bool mutable = fal
                 data: data,
                 displayName:displayDName);
 
-            _slots.Set(slotIndex, info);
+            _slots.SetInitial(slotIndex, info);

```

Comment:
Alloc/setInitial are split because: slotIndex (from Alloc) is part of info.data, which gets passed into SetInitial.